gpg_key.go

```diff
@@ -0,0 +1,184 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// Compile-time proof of interface implementation
+var _ GPGKeys = (*gpgKeys)(nil)
+
+// GPGKeys describes all the GPG key related methods that the Terraform Private Registry API supports.
+//
+// TFE API Docs: https://www.terraform.io/cloud-docs/api-docs/private-registry/gpg-keys
+type GPGKeys interface {
+	// Uploads a GPG Key to a private registry scoped with a namespace.
+	Create(ctx context.Context, options GPGKeyCreateOptions) (*GPGKey, error)
+
+	// Read a GPG key.
+	Read(ctx context.Context, keyID GPGKeyID) (*GPGKey, error)
+
+	// Update a GPG key.
+	Update(ctx context.Context, keyID GPGKeyID, options GPGKeyUpdateOptions) (*GPGKey, error)
+
+	// Delete a GPG key.
+	Delete(ctx context.Context, keyID GPGKeyID) error
+}
+
+// gpgKeys implements GPGKeys
+type gpgKeys struct {
+	client *Client
+}
+
+// GPGKey represents a signed GPG key for a TFC/E private provider.
+type GPGKey struct {
+	ID             string    `jsonapi:"primary,gpg-keys"`
+	AsciiAmor      string    `jsonapi:"attr,ascii-armor"`
+	CreatedAt      time.Time `jsonapi:"attr,created-at,iso8601"`
+	KeyID          string    `jsonapi:"attr,key-id"`
+	Namespace      string    `jsonapi:"attr,namespace"`
+	Source         string    `jsonapi:"attr,source"`
+	SourceURL      *string   `jsonapi:"attr,source-url"`
+	TrustSignature string    `jsonapi:"attr,trust-signature"`
+	UpdatedAt      time.Time `jsonapi:"attr,updated-at,iso8601"`
+}
+
+// GPGKeyID represents the set of identifiers used to fetch a GPG key.
+type GPGKeyID struct {
+	Namespace string
+	KeyID     string
+}
+
+// GPGKeyCreateOptions represents all the available options used to create a GPG key.
+type GPGKeyCreateOptions struct {
+	Type      string `jsonapi:"primary,gpg-keys"`
+	Namespace string `jsonapi:"attr,namespace"`
+	AsciiAmor string `jsonapi:"attr,ascii-armor"`
+}
+
+// GPGKeyCreateOptions represents all the available options used to update a GPG key.
+type GPGKeyUpdateOptions struct {
+	Type      string `jsonapi:"primary,gpg-keys"`
+	Namespace string `jsonapi:"attr,namespace"`
+}
+
+func (s *gpgKeys) Create(ctx context.Context, options GPGKeyCreateOptions) (*GPGKey, error) {
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	req, err := s.client.newRequest("POST", "/api/registry/private/v2/gpg-keys", &options)
```

Comment:
When we talked about these URLs last week, did we say that we would create a new request type for the private registry API endpoint? 🤔 I didn't take notes so I'm not remembering unfortunately.

Reply:
Yep! but these improvements will be made after the fact since they are only internal changes.